ckanext/dgu/bin/fix_mandate.py

```diff
@@ -30,13 +30,22 @@ def command(cls, config_ini, write):
             if 'mandate' in package.extras:
 
                 mandate = package.extras.get('mandate')
-
-                if mandate:
-                    print stats.add('Fixing (with value)', package.name)
-                    package.extras['mandate'] = json.dumps([mandate])
-                else:
-                    print stats.add('Fixing (without value)', package.name)
-                    package.extras['mandate'] = json.dumps([])
+                try:
+                    mandate = json.loads(mandate)
+                    if isinstance(mandate, list):
+                        stats.add('Already list', package.name)
+                    elif isinstance(mandate, basestring):
+                        stats.add('Fixing JSON string', package.name)
+                        package.extras['mandate'] = json.dumps([mandate])
+                    else:
+                        stats.add('Problem JSON', package.name)
+                except:
```

Comment:
Can you catch a particular exception instead of all exceptions? As it is, it could hide all sorts of programming errors.